frontend/packages/dev-console/src/utils/pipeline-actions.tsx

```diff
@@ -18,6 +18,32 @@ export const handlePipelineRunSubmit = (pipelineRun: PipelineRun) => {
     ),
   );
 };
+
+/**
+ * Migrates a PipelineRun from one version to another to support auto-upgrades with old (and invalid) PipelineRuns.
+ *
+ * Note: Each check within this method should be driven by the apiVersion number if the API is properly up-versioned
+ * for these breaking changes. (should be done moving from 0.10.x forward)
+ */
+export const migratePipelineRun = (pipelineRun: PipelineRun): PipelineRun => {
+  let newPipelineRun = pipelineRun;
+
+  const serviceAccountPath = 'spec.serviceAccount';
+  if (_.has(newPipelineRun, serviceAccountPath)) {
```

Comment:
Do we need this `if`, could there be a case, the `if` isn't true but `serviceAccoountName` must be added, in that case the `_.merge` should be put outside the `if`

Reply:
Yes it is important. The goal of this method is to grow with migration paths as tekton evolves. Each if-block will cover a single migration path. 

If you don't need a migration applied the object isn't modified and the weight of this function is next to nothing.